docs/aws_iam_role_crd.yaml

```diff
@@ -0,0 +1,110 @@
+      properties:
+        spec:
+          properties:
+            roleReference:
+              description: |
+                Reference to an AWS IAM role which can either be a role name
+                or a full IAM role ARN.
+              type: string
+              minLength: 3
+            roleSessionDuration:
+              description: |
+                Specify the tole session duration in seconds. defaults to 3600
+                seconds (1 hour). This value must be less than or equal to the
+                `MaxSessionDuration` value of the IAM role.
+              type: integer
+              minimum: 900   # 15 minutes
+              maximum: 43200 # 12 hours
+            roleDefinition:
+              properties:
+                AssumeRolePolicyDocument:
+                  properties:
+                    Statement:
+                      type: array
+                      items:
+                        properties:
+                          Action:
+                            type: string
+                          Effect:
+                            type: string
+                          Principal:
+                            properties:
+                              Service:
+                                type: string
+                ManagedPolicyArns:
+                  type: array
+                  items:
+                    type: string
+                # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html#cfn-iam-role-maxsessionduration
+                MaxSessionDuration:
+                  type: integer
+                  minimum: 3600  # 1 hour
+                  maximum: 43200 # 12 hours
+                Path:
+                  type: string
+                PermissionsBoundary:
+                  type: string
+                Policies:
+                  properties:
+                    PolicyName:
+                      type: string
+                    PolicyDocument:
+                      properties:
+                        Version:
+                          type: string
+                          enum: ["2012-10-17"]
+                        Statement:
+                          type: array
+                          items:
+                            properties:
+                              Effect:
+                                type: string
+                                enum: ["Allow", "Deny"]
+                              Action:
+                                type: array
+                                items:
+                                  type: string
+                              Resource:
+                                type: string
+                  required:
+                  - PolicyName
+                  - PolicyDocument
+        status:
+          properties:
+            observedGeneration:
+              type: integer
+            roleARN:
+              type: string
+            expiration:
+              type: string
```

Comment:
What's the format of this value? Presumably it's a time instant formatted as a string.

Reply:
Yes, RFC3339 formatted string in UTC (I'm not sure if kubectl can somehow show the local time given a timestamp string?)